R/ard_stack_hierarchical.R

```diff
@@ -0,0 +1,403 @@
+#' Stacked Hierarchical ARD Statistics
+#'
+#' @description
+#' `r lifecycle::badge('experimental')`\cr
+#' Use these functions to calculate multiple summaries of nested or hierarchical data
+#' in a single call.
+#'
+#' - `ard_stack_hierarchical()`: Calculates *rates* of events (e.g. adverse events)
+#'   utilizing the `denominator` and `id` arguments to identify the rows in `data`
+#'   to include in each rate calculation.
+#'
+#' - `ard_stack_hierarchical_count()`: Calculates *counts* of events utilizing
+#'   all rows for each tabulation.
+#'
+#' @details
+#' To calculate event rates, the `ard_stack_hierarchical()` function identifies
+#' rows to include in the calculation.
+#' First, the primary data frame is sorted by the columns identified in
+#' the `id`, `by`, and `variables` arguments.
+#'
+#' As the function cycles over the variables specified in the `variables` argument,
+#' the data frame is grouped by `id`, `intersect(by, names(denominator))`, and `variables`
+#' utilizing the last row within each of the groups.
+#'
+#' @inheritParams ard_hierarchical
+#' @inheritParams ard_stack
+#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   Specifies the nested/hierarchical structure of the data.
+#'   The variables that are specified here and in the `include` argument
+#'   will have summary statistics calculated.
+#' @param id ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   argument used to subset `data` to identify rows in `data` to calculate
```

Comment:
I 100% agree it's vague....but I am not sure what the best phrasing would be. It's not exactly 1 row per `variables` per `id` because we're recursively summarizing over the nested `variables` columns. So the first run it may be all the variables in the argument, but the second it's only including a subset of the columns specified in `variables`.

I think I prefer something vague over something that isn't 100% technically correct. But a  change is def warranted. What if we added a note to "see below for details", and then also add more details below. Re-reading the "Details" section, I think it would benefit from a concrete example. ?

Reply:
I added this to the details section:
![image](https://github.com/user-attachments/assets/8938abad-8fc5-48f6-922b-94843563020c)

Reply:
i like it!